pkg/manifests/types.go

```diff
@@ -757,7 +757,7 @@ type RemoteWriteSpec struct {
 	Name string `json:"name,omitempty"`
 	// Defines OAuth2 authentication settings for the remote write endpoint.
 	OAuth2 *monv1.OAuth2 `json:"oauth2,omitempty"`
-	// Defines an optional proxy URL.
```

Comment:
This should be reflected in the docs as well.
(Tell me if we should keep this to another PR)

Reply:
I see several options:
1. keep the current behavior (though it's awkward), if proxyURL is set then use this value, expect if a global HTTP proxy is configured. And it should be explicit in the documentation.
2. Document that the field is deprecated and ignore it.
3. Remove the field (not desired because it could break upgrades).

We could mix 1 and 2: keep the current behavior for a couple of releases and warn users (via the ClusterOperator upgrade condition) that they need to remove it before upgrading to version 4.x (tbd).

Reply:
I still think this is a bug, config fields explicitly provided by users should always have precedence, otherwise it makes to sense to keep them exposed.

Let's go with `1.` for now and see.

Reply:
I agree that it's a bug *but* changing the behavior would be a breaking change for existing users.

Reply:
We can always start warning about the current "unusual" behavior and fix that in a future `4.x.0` making sure it's mentioned in the release notes.

(This "unusual" behavior was introduced in 4.16, so it can also be considered as a breaking change for someone upgrading from 4.15 that used to configure the proxyURL via the configmap because they don't want to go through the cluster-wide one.)

But let's continue this discussion somewhere else :)